toolsrc/src/vcpkg/versions.cpp

```diff
@@ -1,7 +1,26 @@
+#include <vcpkg/base/util.h>
+
 #include <vcpkg/versions.h>
 
+#include <regex>
+
 namespace vcpkg::Versions
 {
+    namespace
+    {
+        Optional<long> as_numeric(const std::string& str)
+        {
+            try
+            {
+                return std::stol(str);
+            }
+            catch (std::exception&)
+            {
+                return nullopt;
+            }
```

Comment:
It would be ideal to avoid exceptions and locale-sensitivity:

```suggestion
            long res = 0;
            for (auto ch : str) {
              if (res > LONG_MAX / 100) return nullopt;
              if (ch < '0' || ch > '9') return nullopt;
              res = res * 10 + (ch - '0');
            }
            return res;
```